sbot/metafeed.go

```diff
@@ -32,94 +37,153 @@ func WithMetaFeedMode(enable bool) Option {
 	}
 }
 
-// MetaFeeds allows managing and publishing to subfeeds of a metafeed.
-type MetaFeeds interface {
-	// CreateSubFeed derives a new keypair, stores it in the keystore and publishes a `metafeed/add` message on the housing metafeed.
-	// It takes purpose whic will be published and added to the keystore, too.
-	// The subfeed will use the format
-	CreateSubFeed(purpose string, format refs.RefAlgo) (refs.FeedRef, error)
+type metaFeedsService struct {
+	rxLog        margaret.Log
+	indexManager ssb.IndexFeedManager
+	users        multilog.MultiLog
+	keys         *keys.Store
+
+	hmacSecret *[32]byte
+}
 
-	// TombstoneSubFeed removes the keypair from the store and publishes a `metafeed/tombstone` message to the feed.
-	// Afterwards the referenced feed is unusable.
-	TombstoneSubFeed(refs.FeedRef) error
+func newMetaFeedService(rxLog margaret.Log, indexManager ssb.IndexFeedManager, users multilog.MultiLog, keyStore *keys.Store, keypair ssb.KeyPair, hmacSecret *[32]byte) (*metaFeedsService, error) {
+	metaKeyPair, ok := keypair.(metakeys.KeyPair)
+	if !ok {
+		return nil, fmt.Errorf("not a metafeed keypair: %T", keypair)
+	}
+	if err := checkOrStoreKeypair(keyStore, metaKeyPair); err != nil {
+		return nil, fmt.Errorf("failed to initialize keystore with root keypair: %w", err)
+	}
 
-	// ListSubFeeds returns a list of all _active_ subfeeds of the housing metafeed
-	ListSubFeeds() ([]SubfeedListEntry, error)
+	return &metaFeedsService{
+		rxLog:        rxLog,
+		indexManager: indexManager,
+		users:        users,
+		hmacSecret:   hmacSecret,
 
-	// Publish works like normal `Sbot.Publish()` but takes an additional feed reference,
-	// which specifies the subfeed on which the content should be published.
-	Publish(as refs.FeedRef, content interface{}) (refs.MessageRef, error)
+		keys: keyStore,
+	}, nil
 }
 
-type SubfeedListEntry struct {
-	Feed    refs.FeedRef
-	Purpose string
+type metadataQuery struct {
```

Comment:
```suggestion
type MetadataQuery struct {
```

Others might want to get to the query data, so let's make this easier.